SA1206 firing incorrectly for `required` in C# 11

Version:
[1.2.0-beta.435](https://www.nuget.org/packages/StyleCop.Analyzers/1.2.0-beta.435)

Repro:
```C#
internal struct SomeStruct
{
    public required int Prop { get; set; }
}
```

Expected:
No warnings

Actual:
```
SA1206: The 'required' modifier should appear before 'public'
```

I tried to create a unit test case, but fail to make the tested code compile:
`error CS1519: Invalid token 'int' in class, record, struct, or interface member declaration`

Could for example be because Microsoft.CodeAnalysis.CSharp.Workspaces 4.3.0-1.final doesn't support this syntax, or maybe we need to configure the tested project to use reference assemblies for .net 7.0?

--- a/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp11/OrderingRules/SA1206CSharp11CodeFixProviderUnitTests.cs
+++ b/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp11/OrderingRules/SA1206CSharp11CodeFixProviderUnitTests.cs
@@ -6,6 +6,7 @@ namespace StyleCop.Analyzers.Test.CSharp11.OrderingRules
     using System.Threading;
     using System.Threading.Tasks;
     using StyleCop.Analyzers.Test.CSharp10.OrderingRules;
+    using StyleCop.Analyzers.Test.Verifiers;
     using Xunit;
     using static StyleCop.Analyzers.Test.Verifiers.StyleCopCodeFixVerifier<
         StyleCop.Analyzers.OrderingRules.SA1206DeclarationKeywordsMustFollowOrder,
@@ -22,6 +23,37 @@ namespace StyleCop.Analyzers.Test.CSharp11.OrderingRules
 
             var expected = Diagnostic().WithLocation(0).WithArguments("file", "unsafe");
             await VerifyCSharpFixAsync(testCode, expected, fixedTestCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        [WorkItem(3527, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3527")]
+        public async Task VerifyRequiredKeywordReorderingInPropertiesAndFieldsAsync()
+        {
+            var testCode = @"
+internal struct SomeStruct
+{
+    required {|#0:public|} int Prop { get; set; }
+    required {|#1:public|} int Field;
+}";
+
+            var fixedCode = @"
+internal struct SomeStruct
+{
+    public required int Prop { get; set; }
+    public required int Field;
+}";
+
+            await new CSharpTest()
+            {
+                ReferenceAssemblies = GenericAnalyzerTest.ReferenceAssembliesNet70,
+                TestCode = testCode,
+                FixedCode = fixedCode,
+                ExpectedDiagnostics =
+                {
+                    Diagnostic().WithLocation(0).WithArguments("public", "required"),
+                    Diagnostic().WithLocation(1).WithArguments("public", "required"),
+                },
+            }.RunAsync(CancellationToken.None).ConfigureAwait(false);
         }
     }
 }

--- a/StyleCop.Analyzers/StyleCop.Analyzers.Test/Verifiers/GenericAnalyzerTest.cs
+++ b/StyleCop.Analyzers/StyleCop.Analyzers.Test/Verifiers/GenericAnalyzerTest.cs
@@ -7,7 +7,6 @@ namespace StyleCop.Analyzers.Test.Verifiers
 {
     using System;
     using System.Collections.Immutable;
-    using System.IO;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.CodeAnalysis;
@@ -22,6 +21,8 @@ namespace StyleCop.Analyzers.Test.Verifiers
         internal static readonly ReferenceAssemblies ReferenceAssembliesNet50;
 
         internal static readonly ReferenceAssemblies ReferenceAssembliesNet60;
+
+        internal static readonly ReferenceAssemblies ReferenceAssembliesNet70;
 
         private static readonly Lazy<IExportProviderFactory> ExportProviderFactory;
 
@@ -44,15 +45,11 @@ namespace StyleCop.Analyzers.Test.Verifiers
             ReferenceAssembliesNet50 = ReferenceAssemblies.Net.Net50.AddPackages(ImmutableArray.Create(
                 new PackageIdentity("Microsoft.CodeAnalysis.CSharp", codeAnalysisTestVersion)));
 
-            ReferenceAssembliesNet60 =
-                new ReferenceAssemblies(
-                    "net6.0",
-                    new PackageIdentity(
-                        "Microsoft.NETCore.App.Ref",
-                        "6.0.0"),
-                    Path.Combine("ref", "net6.0"))
-                .AddPackages(ImmutableArray.Create(
-                    new PackageIdentity("Microsoft.CodeAnalysis.CSharp", codeAnalysisTestVersion)));
+            ReferenceAssembliesNet60 = ReferenceAssemblies.Net.Net60.AddPackages(ImmutableArray.Create(
+                new PackageIdentity("Microsoft.CodeAnalysis.CSharp", codeAnalysisTestVersion)));
+
+            ReferenceAssembliesNet70 = ReferenceAssemblies.Net.Net70.AddPackages(ImmutableArray.Create(
+                new PackageIdentity("Microsoft.CodeAnalysis.CSharp", codeAnalysisTestVersion)));
 
             ExportProviderFactory = new Lazy<IExportProviderFactory>(
                 () =>

--- a/StyleCop.Analyzers/StyleCop.Analyzers/Helpers/ModifierOrderHelper.cs
+++ b/StyleCop.Analyzers/StyleCop.Analyzers/Helpers/ModifierOrderHelper.cs
@@ -69,6 +69,7 @@ namespace StyleCop.Analyzers.OrderingRules
             case SyntaxKind.AsyncKeyword:
             case SyntaxKind.PartialKeyword:
             case SyntaxKind.RefKeyword:
+            case SyntaxKindEx.RequiredKeyword:
                 result = ModifierType.Other;
                 break;
             }

--- a/StyleCop.Analyzers/StyleCop.Analyzers/Lightup/SyntaxKindEx.cs
+++ b/StyleCop.Analyzers/StyleCop.Analyzers/Lightup/SyntaxKindEx.cs
@@ -14,6 +14,7 @@ namespace StyleCop.Analyzers.Lightup
         public const SyntaxKind NotKeyword = (SyntaxKind)8440;
         public const SyntaxKind ManagedKeyword = (SyntaxKind)8445;
         public const SyntaxKind UnmanagedKeyword = (SyntaxKind)8446;
+        public const SyntaxKind RequiredKeyword = (SyntaxKind)8447;
         public const SyntaxKind FileKeyword = (SyntaxKind)8449;
         public const SyntaxKind NullableKeyword = (SyntaxKind)8486;
         public const SyntaxKind EnableKeyword = (SyntaxKind)8487;

